OpenSim/Simulation/Model/Frame.h

```diff
@@ -237,14 +255,24 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Frame, ModelComponent);
     This is mathematically stated as:
         vec_G = X_GF*vec_F  */
     virtual SimTK::Transform
-        calcGroundTransform(const SimTK::State& state) const = 0;
+        calcTransformInGround(const SimTK::State& state) const = 0;
+
+    /** The spatial velocity {omega; v} for this Frame in ground. */
+    virtual SimTK::SpatialVec
+        calcVelocityInGround(const SimTK::State& state) const = 0;
+
+    /** The spatial acceleration {alpha; a} for this Frame in ground */
+    virtual SimTK::SpatialVec
+        calcAccelerationInGround(const SimTK::State& state) const = 0;
```

Comment:
See my related comment about mentioning that Frame takes care of the caching; no need for implementers to cache.